core/DataAccess/ArchiveWriter.php

```diff
@@ -342,7 +342,7 @@ private function checkDoneValueIsOnlyPartialForPluginArchives($doneValue)
     {
         // if the done flag is not like done%.PluginName, then it shouldn't be a partial archive.
         // log a warning.
-        if ($doneValue == self::DONE_PARTIAL && strpos($this->doneFlag, '.') == false) {
+        if ($doneValue === self::DONE_PARTIAL && strpos($this->doneFlag, '.') === false) {
```

Comment:
`$doneValue` might be string though.

```suggestion
        if ((int) $doneValue === self::DONE_PARTIAL && strpos($this->doneFlag, '.') === false) {
```